### ISSUE TITLE
word2id.pkl

### ISSUE THREAD

**Issue by arynas** [Determined22/zh-NER-TF#17]
Did you create word2id.pkl using Glove?

**Comment by cyy0523xc** [comment]
look here: https://github.com/guillaumegenthial/sequence_tagging

**Comment by arynas** [comment]
Ok. Thanks.

**Comment by Determined22** [comment]
Hi,
This is only a mapping from `character` to `character_id`. I create it by using `vocab_build()` in `data.py`. A better choice is `Gensim` package.

**Comment by LordLiang** [comment]
Hi, I try to use  vocab_build() to generate word2id.pkl. Should I set 'corpus_path' be train data or train+dev+test data?  How much should 'min_count' be set？

**Comment by Determined22** [comment]
@LordLiang Hi,
`min_count` is used to determine which words are treated as '<UNK>' (unknown word), I usually set it to five (for small corpus, e.g., this experiment) or 20 (for large corpus). 
I suggest that setting 'corpus_path' to 'train' is reasonable. I have also seen that some codes set it to 'train+dev+test'. This setting is also ok, because word embeddings of 'new words' (i.e., only appear in test set, do not appear in training set) will not be updated during training. 
In practice, using embeddings pretrained on large corpus is a common choice, so I recommend that using `Gensim` for pretraining embeddings and building vocabulary. My code is only a demo, so some details (like vocabulary building) are simplified.

**Comment by tsangsea** [comment]
可以提供保存一个python 2.7的吗

**Comment by ldsyou92** [comment]
@Determined22 Hi，
I‘m using vocab-build() to generate word2id.pkl , but the lentgh of word2id is much smaller than it is in my train_data, so what reasons may cause this problem?

**Comment by 1504020316** [comment]
> @LordLiang Hi,
> `min_count` is used to determine which words are treated as 'UNK' (unknown word), I usually set it to five (for small corpus, e.g., this experiment) or 20 (for large corpus).
> I suggest that setting 'corpus_path' to 'train' is reasonable. I have also seen that some codes set it to 'train+dev+test'. This setting is also ok, because word embeddings of 'new words' (i.e., only appear in test set, do not appear in training set) will not be updated during training.
> In practice, using embeddings pretrained on large corpus is a common choice, so I recommend that using `Gensim` for pretraining embeddings and building vocabulary. My code is only a demo, so some details (like vocabulary building) are simplified.

这个实验的min_count是3吧

**Comment by adelina805105** [comment]
请问vocab_path,embedding_path,source_path分别指的什么呢，如何调用来生成word2id.pkl文件呢
